frontendGraph/src/main/java/com/virtuslab/gitmachete/frontend/graph/repository/RepositoryGraphBuilder.java

```diff
@@ -36,147 +39,135 @@
   public static final IBranchGetCommitsStrategy EMPTY_GET_COMMITS = __ -> List.empty();
 
   public RepositoryGraph build() {
-    return new RepositoryGraph(deriveGraphElements());
+    Tuple2<List<IGraphNode>, List<List<Integer>>> graphData = deriveGraphNodesAndPositionsOfVisibleEdges();
+    return new RepositoryGraph(graphData._1(), graphData._2());
   }
 
-  private List<IGraphElement> deriveGraphElements() {
-    java.util.List<IGraphElement> graphElements = new ArrayList<>();
+  private Tuple2<List<IGraphNode>, List<List<Integer>>> deriveGraphNodesAndPositionsOfVisibleEdges() {
     List<BaseGitMacheteRootBranch> rootBranches = repository.getRootBranches();
+
+    java.util.List<IGraphNode> graphNodes = new ArrayList<>();
+    java.util.List<java.util.List<Integer>> positionsOfVisibleEdges = new ArrayList<>(
+        Collections.nCopies(rootBranches.size(), new SmartList<>()));
```

Comment:
Hmm wait doesn't `Collections.nCopies` just copy the very same element instance `n` times? :thinking:
So in this case, adding a sub-element to any element of `nCopies`'s result would effectively add this very sub-element to all other elements...
If that's indeed the case, pls find out if there's a branch layout that this would fail on

Reply:
TBH I'd add `Collections.nCopies` to Checkstyle's forbidden `Regexp`s while you're here...

Reply:
You are a real bug hunter 

![image](https://user-images.githubusercontent.com/19799111/79737881-46d5af00-82fc-11ea-8d5a-b403d2aa6787.png)

Reply:
For more than 2 root branches this happens. I think that some visible edges may cover each other. 

>java.lang.IndexOutOfBoundsException: get(6) on List of length 4
	at io.vavr.collection.List.get(List.java:917)
	at com.virtuslab.gitmachete.frontend.graph.repository.RepositoryGraph.getGraphNode(RepositoryGraph.java:48)
...

Reply:
That's why I insist on immutable collections :grammar-nazi: